lib/get-config.js

```diff
@@ -283,6 +286,51 @@ function processIgnores(config) {
   config.ignore = config.ignore.map((pattern) => micromatch.matcher(pattern));
 }
 
+/**
+ * we were passed a rule, add the rule being passed in, to the config.
+ * ex:
+ * rule:severity
+ * no-implicit-this:["error", { "allow": ["some-helper"] }]
+ */
+function getRuleFromString(rule) {
+  const indexOfSeparator = rule.indexOf(':') + 1;
+
+  // we have to split based on the index of the first instance of the separator because the separator could exist in the second half of the rule
+  const name = rule.substring(0, indexOfSeparator - 1);
+  let severity = rule.substring(indexOfSeparator);
+  let config = true;
+
+  // if we get passed an object that looks like no-implicit-this:["error", { "allow": ["some-helper"] }] we should split it accordingly
+  try {
+    const options = JSON.parse(severity);
+
+    severity = options[0];
+    config = options[1];
+  } catch (ex) {
+    // noop
```

Comment:
Is this really a `noop`? Doesn't this mean that the user passed us garbage data? Assuming that is true, we really should throw an error here (just a nicer formatted one).

Reply:
We have to check if the user passed json or not, how would we do that without a try catch?

Reply:
YA, but the issue is that if the user **didn't** pass us JSON it is actually a failure scenario. Right?

Reply:
AFAICT the only thing we expect _other than_ JSON is three literal strings (`warn`,`error`,`off`) which we can just check for then fallback to `JSON.parse`. We'd still need the try/catch in order to provide an error that is actually helpful (vs the default `JSON.parse` error).

Reply:
removed this in the most recent update.